symfony/plugins/orangehrmAdminPlugin/Dao/WorkShiftDao.php

```diff
@@ -19,106 +19,171 @@
+        return $paginator->count();
+    }
 
-            // @codeCoverageIgnoreStart
-        } catch (Exception $e) {
-            throw new DaoException($e->getMessage(), $e->getCode(), $e);
-        }
-        // @codeCoverageIgnoreEnd
+    /**
+     * @param WorkShift $workShift
+     * @param int[] $empNumbers
+     * @return WorkShift
+     * @throws TransactionException
+     */
+    public function saveWorkShift(WorkShift $workShift, array $empNumbers): WorkShift
+    {
+        $this->beginTransaction();
+        try{
+            $this->persist($workShift);
+            $this->commitTransaction();
+            if (count($empNumbers) > 0) {
+                // this function will invoke only if the array have some values
+                $this->saveEmployeeWorkShift($empNumbers, $workShift);
+                return $workShift;
+            }
+            }
+            catch (Exception $e){
+                $this->rollBackTransaction();
+                throw new TransactionException($e);
+            }
+        return $workShift;
     }
 
-    public function getWorkShiftEmployeeList() {
-        // TODO
-        try {
-            $q = Doctrine_Query :: create()
-                    ->from('EmployeeWorkShift');
-            return $q->execute();
-        } catch (Exception $e) {
-            throw new DaoException($e->getMessage());
+    /**
+     * @param int[] $empNumbers
+     * @param WorkShift $workShift
+     */
+    public function saveEmployeeWorkShift(array $empNumbers, WorkShift $workShift): void
+    {
+        foreach ($empNumbers as $empNumber) {
+            $employeeWorkShift = new EmployeeWorkShift();
+            $employee = $this->getRepository(Employee::class)->find($empNumber);
+            $employeeWorkShift->setWorkShift($workShift);
+            $employeeWorkShift->setEmployee($employee);
+            $this->getEntityManager()->persist($employeeWorkShift);
         }
+        $this->getEntityManager()->flush();
     }
 
-    public function getWorkShiftEmployeeIdList() {
-        // TODO
-        try {
-            $q = Doctrine_Query :: create()
-                    ->select('emp_number')
-                    ->from('EmployeeWorkShift');
-
-            $employeeIds = $q->execute(array(), Doctrine_Core::HYDRATE_SINGLE_SCALAR);
-
-            if (is_string($employeeIds)) {
-                $employeeIds = array($employeeIds);
+    /**
+     * @param WorkShift $workShift
+     * @param int[] $empNumbers
+     * @return WorkShift
+     */
+    public function updateWorkShift(WorkShift $workShift, array $empNumbers): WorkShift
+    {
+        $existingEmployees = $this->getEmployeeListByWorkShiftId($workShift->getId());
+        $idList = array();
+        foreach ($existingEmployees as $x => $existingEmployee) {
+            $id = $existingEmployee->getEmpNumber();
+            if (!in_array($id, $empNumbers)) {
+                $this->deleteExistingEmployees($workShift->getId(), $id);
```

Comment:
we can optimize this to delete employees using one query, since workshift id not changing